lib/web_ui/test/engine/pointer_binding_test.dart

```diff
@@ -3087,6 +3087,385 @@ void testMain() {
+        'clientY': testElement.getBoundingClientRect().y,
+      }
+    );
+
+    binding.clickDebouncer.onClick(click, 42, false);
+    expect(
+      reason: 'When tappable declares that it is not listening to click events '
+              'the debouncer flushes the pointer events to the framework and '
+              'lets it sort it out.',
+      pointerPackets,
+      <ui.PointerChange>[
+        ui.PointerChange.add,
+        ui.PointerChange.down,
+      ],
+    );
+    expect(semanticsActions, isEmpty);
+  });
+
+  testWithSemantics('Dedupes click if pointer down/up flushed recently', () async {
+    expect(EnginePlatformDispatcher.instance.semanticsEnabled, true);
+    expect(binding.clickDebouncer.isDebouncing, false);
+
+    final DomElement testElement = createDomElement('flt-semantics');
+    testElement.setAttribute('flt-tappable', '');
+    flutterViewEmbedder.semanticsHostElement!.appendChild(testElement);
+
+    testElement.dispatchEvent(context.primaryDown());
+
+    // Simulate the user holding the pointer down for some time before releasing,
+    // such that the pointerup event happens close to timer expiration. This
+    // will create the situation that the click event arrives just after the
+    // pointerup is flushed. Forwarding the click to the framework would look
+    // like a double-click, so the click event is deduped.
+    await Future<void>.delayed(const Duration(milliseconds: 190));
+
+    testElement.dispatchEvent(context.primaryUp());
+    expect(binding.clickDebouncer.isDebouncing, true);
+    expect(
+      reason: 'Timer has not expired yet',
+      pointerPackets, isEmpty,
+    );
+
+    // Wait for the timer to expire to make sure pointer events are flushed.
+    await Future<void>.delayed(const Duration(milliseconds: 11));
+
+    expect(
+      reason: 'Queued up events should be flushed to the framework because the '
+              'time expired before the click event arrived.',
+      pointerPackets,
+      <ui.PointerChange>[
+        ui.PointerChange.add,
+        ui.PointerChange.down,
+        ui.PointerChange.up,
+      ],
+    );
+
+    final DomEvent click = createDomMouseEvent(
+      'click',
+      <Object?, Object?>{
+        'clientX': testElement.getBoundingClientRect().x,
+        'clientY': testElement.getBoundingClientRect().y,
+      }
+    );
+    binding.clickDebouncer.onClick(click, 42, true);
+
+    expect(
+      reason: 'Because the DOM click event was deduped.',
+      semanticsActions,
+      isEmpty,
+    );
+  });
+
+
+  testWithSemantics('Forwards click if enough time passed after the last flushed pointerup', () async {
+    expect(EnginePlatformDispatcher.instance.semanticsEnabled, true);
+    expect(binding.clickDebouncer.isDebouncing, false);
+
+    final DomElement testElement = createDomElement('flt-semantics');
+    testElement.setAttribute('flt-tappable', '');
+    flutterViewEmbedder.semanticsHostElement!.appendChild(testElement);
```

Comment:
From what I've seen, `flt-tappable` always goes in a `flt-semantics`? Can it be the case that we have something like:

```
flt-semantics
 ┗ some-elements...
    ┗ [flt-tappable]
```

(So the flt-tappable is not directly on the element of tag flt-semantics?)

Reply:
No, it is guaranteed by the `Tappable` role manager that the `flt-tappable` attribute is set on the `flt-semantics` element itself. However, I'm realizing that I don't have a test for that. Adding a test.